### ISSUE TITLE
Seeing new "Fetching resource at" error randomly with Cypress after upgrading

### ISSUE THREAD

**Issue by kaiyoma** [cypress-io/cypress#15434]
### Current behavior
After upgrading to Cypress 6.6.0 (from Cypress 6.2.x), we are now seeing a new random error pop up in our tests:

```
Error: The following error originated from your test code, not from Cypress.

  > Fetching resource at '/__cypress/tests?p=<path to test>' failed

When Cypress detects uncaught errors originating from your test code it will automatically fail the current test.

Cypress could not associate this error to any specific test.

We dynamically generated a new test to display this failure.
    at XMLHttpRequest.xhr.onerror (http://localhost:4001/__cypress/runner/cypress_runner.js:175913:14)
throw err;
```

### Desired behavior
This shouldn't happen.

### Test code to reproduce
No idea how to repro this. It happens randomly and it's really not clear why Cypress is failing.

### Versions

Current version is 6.6.0, last known working version was 6.2.x.

**Comment by jennifer-shehane** [comment]
Can you provide a reproducible example? 

Does this happen locally during cypress open? Only in CI? Any more information to recreate will be the only way we can investigate. Thanks.

**Comment by kaiyoma** [comment]
Unfortunately this is happening infrequently and randomly, so I don't have a repro example or steps yet. I'll try running a bunch of tests today via `cypress open` and I'll report back if I see this error there.

**Comment by kaiyoma** [comment]
I just saw this in `cypress open`. Again, it happened randomly, with code that normally works fine. Re-running the test worked fine, as expected.

**Comment by MaximeHeckel** [comment]
@jennifer-shehane my team has been facing the same issue recently.
It only happens on CI (cypress github action latest version) for us but like @kaiyoma says, it's totally random and the output from both the cypress runner and the cypress dashboard is not very helpful.

It's been impacting us for about a week now, we've tried to upgrade to the latest version but it's still happening.

Right now we're on Cypress 6.8.0, but we've had issues since Cypress 6.2.1

**Comment by christian-miljkovic** [comment]
> @jennifer-shehane my team has been facing the same issue recently.
> It only happens on CI (cypress github action latest version) for us but like @kaiyoma says, it's totally random and the output from both the cypress runner and the cypress dashboard is not very helpful.
> 
> It's been impacting us for about a week now, we've tried to upgrade to the latest version but it's still happening.
> 
> Right now we're on Cypress 6.8.0, but we've had issues since Cypress 6.2.1
> 
> UPDATE: The solution that's brought up in #7764 worked on our end.

The proposed solution is installing `uws` which is no longer supported on npm correct? I checked and NPM doesn't have `uws` anymore due to a dispute with the author... also not sure how installing that specific web sockets library would help.  

I am also running into this problem while running cypress in a docker container in a concourse CI pipeline.

**Comment by bitttttten** [comment]
I first ran into this issue on CI with the github action, although now I am reproduce it running locally with `cypress run`.

TL;DR: it seems that always the first test passes, and subsequent tests fail.

I tried adding `uws` with `yarn add uws` which did not work for me. I even tried installing uws@100 and uws@99 even though they are deprecated. Initially I thought it was because I was using a service worker that intercepts network requests, as I use [msw](https://github.com/mswjs/msw) to mock some requests out when running cypress. Although it doesn't seem to be the case unless all the people in this thread are doing that 🙃  I also tried creating a repro in another repo, including added msw, although I couldn't get it to fail.

This is the output I get. I first start the app in one terminal, it runs on port 3000.

```sh
$ yarn start
```

Then I run cypress:

```sh
$ CYPRESS_BASE_URL=http://localhost:3000/ && npx cypress run
yarn run v1.22.10
$ cypress run
Missing baseUrl in compilerOptions. tsconfig-paths will be skipped

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:    6.8.0                                                                              │
  │ Browser:    Electron 87 (headless)                                                             │
  │ Specs:      5 found (auth.spec.ts, home.spec.ts, navigation.spec.ts, profile.spec.ts, project. │
  │             spec.ts)                                                                           │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  auth.spec.ts                                                                    (1 of 5)


  auth
    ✓ registers user and logs in and can visit user profile page (8999ms)
    ✓ logs in with admin account (3949ms)
    ✓ logs into page from url (3938ms)


  3 passing (17s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        3                                                                                │
  │ Passing:      3                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     16 seconds                                                                       │
  │ Spec Ran:     auth.spec.ts                                                                     │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  home.spec.ts                                                                    (2 of 5)


  1) An uncaught error was detected outside of a test

  0 passing (268ms)
  1 failing

  1) An uncaught error was detected outside of a test:
     Error: The following error originated from your test code, not from Cypress.

  > Fetching resource at '/__cypress/tests?p=cypress/support/index.ts' failed

When Cypress detects uncaught errors originating from your test code it will automatically fail the current test.

Cypress could not associate this error to any specific test.

We dynamically generated a new test to display this failure.
      at XMLHttpRequest.xhr.onerror (http://localhost:3000/__cypress/runner/cypress_runner.js:175990:14)


```

(I cancelled the test here when the second test failed, otherwise it gets to the end and:

```sh
  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  auth.spec.ts                             00:17        3        3        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  home.spec.ts                             263ms        1        -        1        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  navigation.spec.ts                       256ms        1        -        1        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  profile.spec.ts                          262ms        1        -        1        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  project.spec.ts                          260ms        1        -        1        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖  4 of 5 failed (80%)                      00:18        7        3        4        -        -  
```

4/5 tests fail with the same error.

What's odd is that if I exclude `auth.spec.ts`:

```sh
  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  home.spec.ts                             00:12        2        2        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  navigation.spec.ts                       280ms        1        -        1        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  profile.spec.ts                          249ms        1        -        1        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  project.spec.ts                          260ms        1        -        1        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖  3 of 4 failed (75%)  
```

Then again the first test passes.

I can also just run one test and it'll pass:

```sh
CYPRESS_BASE_URL=http://localhost:3000/ && npx cypress run --spec 'cypress/integration/project.spec.ts'
Missing baseUrl in compilerOptions. tsconfig-paths will be skipped

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:    6.8.0                                                                              │
  │ Browser:    Electron 87 (headless)                                                             │
  │ Specs:      1 found (project.spec.ts)                                                          │
  │ Searched:   cypress/integration/project.spec.ts                                                │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  project.spec.ts                                                                 (1 of 1)


  project
    ✓ views a project (7472ms)


  1 passing (8s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     7 seconds                                                                        │
  │ Spec Ran:     project.spec.ts                                                                  │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  project.spec.ts                          00:07        1        1        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        00:07        1        1        -        -        -  
```

I am still digging around, hopefully I can find something :)

**Comment by nselikoff** [comment]
I'm seeing this as well and suspect it has something to do with msw.

My setup:
- cypress 6.8.0
- cypress testing library 7.0.4
- msw 0.27.1
- start-server-and-test 1.12.1
- create react app
- craco

I'm running my dev server and cypress via `npm run test:e2e` which calls `start-server-and-test start 3000 cypress`

In my react app's src/index.js, this is how I conditionally start the msw worker:
```
if (process.env.NODE_ENV === "development") {
  const { worker } = require("./mocks/browser");
  worker.start().then(() => renderApp());
} else {
  renderApp();
}
```

With this setup, the cypress test runs the first time, but fails with the "Fetching resource at" error on subsequent runs. This happens whether I'm clicking the Run All Tests button in the cypress UI, or saving a file (which triggers recompile, and triggers the test to run again).

When I disable the msw worker (i.e. just call `renderApp()`), the cypress test doesn't fail after the first run.

**Comment by nselikoff** [comment]
If you're using msw, try editing your (autogenerated) mockServiceWorker.js file as described here: https://github.com/mswjs/msw/issues/643#issuecomment-797505097

> Updating line 117 on mockServiceWorker.js to this seems to resolve the problem
>
>  if (client && activeClientIds.has(client.id)) {

@bitttttten

**Comment by bitttttten** [comment]
Ahh so it might be related to msw. Thanks!

**Comment by kaiyoma** [comment]
We don't have `msw` in our project, so it seems this issue doesn't necessarily require it.

**Comment by MaximeHeckel** [comment]
yup same on our end we do not use msw

**Comment by jithinjosejacob** [comment]
i am getting the same error in ci inside docker containers,locally it works fine, as said above, first test passes and then next text onwards it fails with above issue. i am in 6.8.0 version

**Comment by LeBenLeBen** [comment]
I had the same issue and it was related to msw as well. I tried your solution @nselikoff and it fixed Cypress but then tests using msw would fail because the mock wasn’t behaving properly. The workaround proposed in the following issue is both fixing Cypress and allow my tests to continue passing: https://github.com/mswjs/msw/issues/685

**Comment by MaximeHeckel** [comment]
Are there any updates on this or any recommendation on how to workaround this in the meantime? It's impacting our team quite a lot

**Comment by jithinjosejacob** [comment]
Any update on this issue,or are there any workarounds for this issue
